### ISSUE TITLE
Inserted GameData hook crashing on startup

### ISSUE THREAD

**Issue by void513** [keepcalm/BukkitForge#570]
Server crashes on start using BukkitForge-1.4.7-255.build
2013-03-06 18:48:23 [INFO] [STDOUT] THIS SERVER IS RUNNING BUKKITFORGE null-null. JUST IN CASE SOMEONE ASKS!
2013-03-06 18:48:24 [INFO] [STDOUT] Inserted GameData hook
2013-03-06 18:48:24 [INFO] [STDERR] Exception in thread "main" java.lang.NoClassDefFoundError: gm
2013-03-06 18:48:24 [INFO] [STDERR]     at java.lang.Class.getDeclaredMethods0(Native Method)
2013-03-06 18:48:24 [INFO] [STDERR]     at java.lang.Class.privateGetDeclaredMethods(Unknown Source)
2013-03-06 18:48:24 [INFO] [STDERR]     at java.lang.Class.getDeclaredMethod(Unknown Source)
2013-03-06 18:48:24 [INFO] [STDERR]     at cpw.mods.fml.relauncher.ReflectionHelper.findMethod(ReflectionHelper.java:170)
2013-03-06 18:48:24 [INFO] [STDERR]     at cpw.mods.fml.relauncher.FMLRelauncher.relaunchServer(FMLRelauncher.java:138)
2013-03-06 18:48:24 [INFO] [STDERR]     at cpw.mods.fml.relauncher.FMLRelauncher.handleServerRelaunch(FMLRelauncher.java:33)
2013-03-06 18:48:24 [INFO] [STDERR]     at net.minecraft.server.MinecraftServer.main(MinecraftServer.java:1376)
2013-03-06 18:48:24 [INFO] [STDERR] Caused by: java.lang.ClassNotFoundException: gm
2013-03-06 18:48:24 [INFO] [STDERR]     at cpw.mods.fml.relauncher.RelaunchClassLoader.findClass(RelaunchClassLoader.java:185)
2013-03-06 18:48:24 [INFO] [STDERR]     at java.lang.ClassLoader.loadClass(Unknown Source)
2013-03-06 18:48:24 [INFO] [STDERR]     at java.lang.ClassLoader.loadClass(Unknown Source)
2013-03-06 18:48:24 [INFO] [STDERR]     ... 7 more
2013-03-06 18:48:24 [INFO] [STDERR] Caused by: java.lang.RuntimeException: java.lang.ClassNotFoundException: ig
2013-03-06 18:48:24 [INFO] [STDERR]     at org.objectweb.asm.ClassWriter.getCommonSuperClass(Unknown Source)
2013-03-06 18:48:24 [INFO] [STDERR]     at org.objectweb.asm.ClassWriter.a(Unknown Source)
2013-03-06 18:48:24 [INFO] [STDERR]     at org.objectweb.asm.Frame.a(Unknown Source)
2013-03-06 18:48:24 [INFO] [STDERR]     at org.objectweb.asm.Frame.a(Unknown Source)
2013-03-06 18:48:24 [INFO] [STDERR]     at org.objectweb.asm.MethodWriter.visitMaxs(Unknown Source)
2013-03-06 18:48:24 [INFO] [STDERR]     at org.objectweb.asm.tree.MethodNode.accept(Unknown Source)
2013-03-06 18:48:24 [INFO] [STDERR]     at org.objectweb.asm.tree.MethodNode.accept(Unknown Source)
2013-03-06 18:48:24 [INFO] [STDERR]     at org.objectweb.asm.tree.ClassNode.accept(Unknown Source)
2013-03-06 18:48:24 [INFO] [STDERR]     at com.eoware.asm.asmagic.AsmagicClassTransformer.transformClass(AsmagicClassTransformer.java:88)
2013-03-06 18:48:24 [INFO] [STDERR]     at com.eoware.asm.asmagic.AsmagicClassTransformer.transform(AsmagicClassTransformer.java:29)
2013-03-06 18:48:24 [INFO] [STDERR]     at keepcalm.mods.bukkit.asm.transformers.BukkitAsmagicTransformer.transform(BukkitAsmagicTransformer.java:67)
2013-03-06 18:48:24 [INFO] [STDERR]     at cpw.mods.fml.relauncher.RelaunchClassLoader.runTransformers(RelaunchClassLoader.java:228)
2013-03-06 18:48:24 [INFO] [STDERR]     at cpw.mods.fml.relauncher.RelaunchClassLoader.findClass(RelaunchClassLoader.java:173)
2013-03-06 18:48:24 [INFO] [STDERR]     ... 9 more

**Comment by eagl3s1ght** [comment]
This happens on Tekkit Lite 0.6.0, more specifically Build 534 of Forge.

Here's a conversation with IceWolf (Tekkit Lite dev/author) over IRC:
http://puu.sh/2dqb3/95a37ad0d4

tl;dr: Bukkitforge 255 is incompatible with Forge 534 / Tekkit Lite 0.6.0

**Comment by void513** [comment]
ok i fig out the prob it was TreeCapitator cause the prob after i removed it it loaded right up.so there is a issue between bukkit forge and TreeCapitator thats causeing this error

**Comment by dequis** [comment]
Confirmed that it doesn't apply to vanilla + forge 534 + bukkitforge.

**Comment by dequis** [comment]
Confirmed by myself and several people, `[1.4.6]TreeCapitator.Forge.1.4.6.r07.Uni.CoreMod.jar` is causing the conflict. The last referenced ticket says that 252 isn't affected.

**Comment by tonnic** [comment]
Soooo is this a TreeCapitator fix or bukkitforge fix?

**Comment by dequis** [comment]
?PARSE ERROR

I haven't checked which side is doing things wrong, if that's what you're asking.

**Comment by NotoriousPyro** [comment]
I also had this problem. It disappears when reverting to BukkitForge 252.

**Comment by Erudition** [comment]
Yes, using 257 now, treecapitator removal fixes the problem! thanks!

**Comment by void513** [comment]
version [1.4.6]TreeCapitator.Forge.1.4.6.r05.Uni.CoreMod works with BukkitForge-1.4.7-10.

**Comment by dequis** [comment]
@void513 great find! Thanks!

Confirmed it myself, r05 works, r06 doesn't. According to the downloads in [this page](http://bspk.rs/MC/TreeCapitator_Forge/index.html) the release dates of each were `1.4.6.r06 - 05-Feb-2013` and `1.4.6.r05 - 31-Jan-2013` (just for reference)

The diffs of the changes done in that release are available here:
https://github.com/bspkrs/TreeCapitator/commits/1.4.6.r06

So... it got reduced to a regression test between two or three commits? r05 isn't git tagged but there's a "cleanup commit" that might or might not be included there. Also not quite sure why BF 252 is fine, and or which part of the commits break it.

It's also the first time i see they are open source and yeah, they do ASM hax.

---

For users, to recap: Either downgrade to BF 252, downgrade to treecapitator 1.4.6.r05, or remove treecapitator. Any of these options workaround the issue. The treecapitator client side doesn't seem to care about the server not having the corresponding mod.

**Comment by tonnic** [comment]
I tried to downgrade to 1.4.6.r05, but I don't think the bspkrsCore works... I tried both pre1.5 versions.  Anyone get this to work with 1.4.7 and downgrading?

**Comment by void513** [comment]
I got to work with [1.4.6]TreeCapitator.Forge.1.4.6.r05.Uni.CoreMod and [1.4.7]bspkrsCorev1.02.zip.just be sure to put [1.4.6]TreeCapitator.Forge.1.4.6.r05.Uni.CoreMod in coremods folder and put [1.4.7]bspkrsCorev1.02.zip into mods folder.when u log in to server it might tell u  it was unable to inject [1.4.6]TreeCapitator.Forge.1.4.6.r05.Uni.CoreMod  but it still seems to work.
